src/fides/core/dataset.py

```diff
@@ -409,10 +414,23 @@ def get_snowflake_schemas(
             metadata[schema] = {}
             table_cursor = engine.execute(text(f'SHOW TABLES IN "{schema}"'))
             db_tables = [row[1] for row in table_cursor]
-            for table in db_tables:
-                column_cursor = engine.execute(
-                    text(f'SHOW COLUMNS IN "{schema}"."{table}"')
-                )
-                columns = [row[2] for row in column_cursor]
+            number_of_threads = 8 if len(db_tables) > 250 else 4
+            fields = Parallel(n_jobs=number_of_threads, backend="threading")(
```

Comment:
not trying to make things more difficult, but could you use the builtin [`multiprocessing`](https://docs.python.org/3/library/multiprocessing.html) package to get the same functionality here? (i think just a `Pool.map` may be able to get you want you need?)

granted - i'm not a python parallelization expert, and i don't know a whole lot about `joblib`. it also seems pretty lightweight, so i don't think it's necessarily a problem. but i do generally think we should have compelling reasons to stray away from the standard library packages -- maybe there are some here, and i just don't know 'em!

Reply:
to be honest, I knew we used this as a fairly simple way to improve performance in Fidesplus today with Classify so considered it relatively low-risk here as well even though it is introducing something "new" (especially taking the required turn-time component into account)

Reply:
Will take a look now to trial `multiprocessing`! Would be nice to not have to deal with a new addition if we can

Reply:
ah ok totally fine if this is tried and true - doesn't seem like a heavyweight dep and yeah it's already in the plus image.

it's always nice to try and be as slim as possible, especially here in OSS where this gets packaged into the python package. but it seems like minimal impact in this case and not worth a big struggle to switch over!

Reply:
absolutely, would love to get rid of the requirement in both if possible!